# Мини-проект: аналитика в авиакомпании

**Задачи**
1. Импортировать файлы.
2. Изучить данные в них.
3. Проверить типы данных на корректность.
4. Выбрать топ-10 городов по количеству рейсов.
5. Построить графики: модели самолетов и количество рейсов, города и количество рейсов, топ-10 городов и количество рейсов.
6. Сделать выводы по каждому из графиков, пояснить результат.

### Импорт библиотек

In [1]:
import pandas as pd

from pylab import rcParams
rcParams['figure.figsize'] = 8, 5

from plotly.offline import download_plotlyjs, init_notebook_mode, plot, iplot
import plotly
import plotly.graph_objs as go
from plotly.subplots import make_subplots
import plotly.express as px

init_notebook_mode(connected=True)

### Загрузка данных

In [6]:
model_flights = pd.read_csv('/datasets/query_1.csv')
city_flights = pd.read_csv('//datasets/query_3.csv')

In [7]:
model_flights.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8 entries, 0 to 7
Data columns (total 2 columns):
model             8 non-null object
flights_amount    8 non-null int64
dtypes: int64(1), object(1)
memory usage: 256.0+ bytes


In [8]:
model_flights.describe()

flights_amount
count        8.000000
mean      2035.625000
std       1965.203947
min        300.000000
25%        605.250000
50%        795.000000
75%       4250.250000
max       4557.000000

In [9]:
city_flights.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 101 entries, 0 to 100
Data columns (total 2 columns):
city               101 non-null object
average_flights    101 non-null float64
dtypes: float64(1), object(1)
memory usage: 1.7+ KB


In [10]:
city_flights.describe()

average_flights
count       101.000000
mean          5.494189
std          13.119790
min           1.000000
25%           2.000000
50%           3.000000
75%           5.870968
max         129.774194

### Предобработка данных

In [17]:
#В столбцах 'average_flights' и 'flights_amount', 
# излишнее количество знаков после запятой, я думаю хорошее решение округлить до 2-х знаков
city_flights['average_flights'] = city_flights['average_flights'].round(2)
model_flights['flights_amount'] = model_flights['flights_amount'].round(2)

### Анализ данных

In [12]:
fig = px.bar(model_flights.sort_values(by='flights_amount', ascending=True), 
             x='flights_amount', 
             y='model',  
             color = 'flights_amount',
             height=400, 
             orientation='h',
             title = 'Какие самолёты летати в сентябре 2018 года?',
             labels={'model':'Модель самолёта', 'flights_amount':'Количество рейсов'})
fig.show()

In [25]:
fig = px.bar(city_flights.sort_values(by='average_flights', ascending=True).tail(10), 
             x='average_flights', 
             y='city',  
             color = 'average_flights',
             height=400, 
             orientation='h',
             title = 'Топ 10 городов по количеству рейсов в день (сентябрь 2018 года)',
             labels={'city':'Город', 'average_flights':'Количество рейсов в день'})
fig.show()

### Вывод

1. В сентябре 2018 года самый больше всего рейсов совершили Cessna 208, CRJ-208 и SuperJet 100.
2. Из тех самолётов которые представлены в данных, реже всех в небо поднимался Boieng 777-300. 
3. Очевидно, но большенство рейсов вылетает и прилетает в Москву, в среднем по 130 рейсов в день (в сентябре 2018 года), почти в 4 раза меньше рейсов обслуживает северная столица.
4. Новосибирск и Красноярск центры сибирской пассажиской авиации, в среднем обслуживают по 17 и 11 рейсов в день, соответственно.
5. В моём родном городе Ростове в новом аэропорту "Платов" обслуживаю в среднем по 10 рейсов в день. 
6. В среднем крупные города европейской части России обслуживаю по 10 рейсов в день.
7. Небольшие сибирские, уральские города принимают по 1-2 рейса в день.